src/app/components/search/CustomSliderComponent.tsx

```diff
@@ -0,0 +1,145 @@
+import HorizontalRuleIcon from "@mui/icons-material/HorizontalRule";
+import { Box, Slider, Stack, Typography } from "@mui/joy";
+import React, { useEffect } from "react";
+
+import MinMaxInputComponent from "./MinMaxInputComponent";
+
+function valueText(value: number) {
+  return `${value}`;
+}
+
+interface CustomSliderComponentProps {
+  onSliderChange: (values: [number, number]) => void;
+  defaultRangeValue: [number, number];
+  sliderRange: number[];
+  marks?: { value: number; label: string }[];
+  showMinMaxInputs?: boolean;
+  label: string;
+}
+
+export default function CustomSliderComponent({
+  onSliderChange,
+  defaultRangeValue,
+  sliderRange,
+  marks,
+  label,
+  showMinMaxInputs = true,
+}: CustomSliderComponentProps) {
+  const [range, setRange] = React.useState<number[]>([
```

Comment:
Try to use `useMemo` here to compute the range based off of `minValue` and `maxValue` rather than using `useEffect`